cmd/nerdctl/run_network.go

```diff
@@ -164,6 +168,57 @@ func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]
 				ports = append(ports, pm...)
 			}
 		}
+	case nettype.Container:
+		if runtime.GOOS != "linux" {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can only works on linux")
+		}
+		if rootlessutil.IsRootless() {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can't run rootless")
+		}
+		if len(netSlice) > 1 {
+			return nil, nil, "", nil, fmt.Errorf("only one network allowed using '--network=container:<container>'")
+		}
+		network := strings.Split(netSlice[0], ":")
+		if len(network) != 2 {
+			return nil, nil, "", nil, fmt.Errorf("invalid network: %s, should be \"container:<id|name>\"", netSlice[0])
+		}
+		containerName := network[1]
+		client, ctx, cancel, err := newClient(cmd)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		defer cancel()
+		f := &containerInspector{
+			mode: "native",
+		}
+		walker := &containerwalker.ContainerWalker{
+			Client:  client,
+			OnFound: f.Handler,
```

Comment:
Don't use `containerInspector.Handler`
```suggestion
			OnFound: func(ctx context.Context, found containerwalker.Found) error {
			        if found.MatchCount > 1 {
			                return fmt.Errorf("multiple IDs found with provided prefix: %s.", containerName)
			        }
			        n, err := containerinspector.Inspect(ctx, found.Container)
			        if err != nil {
			                return err
			        }
			        if n.Process.Status.Status != containerd.Running {
			                return fmt.Errorf("cannot join network of a non running container: %s", found.Container.ID()) 
			        }
			        return nil
			},
```